### ISSUE TITLE
In APE_Server-1.1.2-i386.bin, on running aped getting error.. Failed to load ../modules/lib/libmod_spidermonkey.so

### ISSUE THREAD

**Issue by symsun** [APE-Project/APE_Server#56]
In APE_Server-1.1.2-i386.bin, on running aped getting error.

[Module] Failed to load ../modules/lib/libmod_spidermonkey.so [Invalid library](libmysqlclient_r.so.16: cannot open shared object file: No such file or directory)

I am currently using APE Server 1.1, I don't have any issues in using the current version. After so many years found a new version. Thought there will be improvements but i could not upgrade to latest stable version. 

I don't connect to MySQL / execute any MySQL query in my APE Client but why the spider monkey is looking for mysql client.

Can you please help me to resolve this issue.

**Comment by lcharette** [comment]
Probably the compiled library was build on a system with MySQL support available. Try installing the mysql lib package (`sudo apt-get install libmysqlclient-dev`) or build from source (See: https://github.com/APE-Project/APE_Server/wiki/APE-Server-install#installing-from-source-or-git)

I'll see when I get a minute if I can upload a binary without MySQL support. A more permanent fix would be to have APE check if MySQL is available while the server is starting and load the correct module. If someone knows how to code that :sweat_smile:

**Comment by symsun** [comment]
Tried downloading again "http://ape-project.org/download/stable/APE_Server-1.1.2-i386.bin.zip" and "http://ape-project.org/download/stable/APE_JSF-1.1.0.tar.gz".

Now the server is running. But nothing works in "APE JSF 1.1".
Also on "Launch test" it shows as "APE JSF Version 1.0". See below launch test debug output,

Debug output:
Running test : Init
Running test : Loading Client
Running test : Loading config
Config values are :

identifier : ape
init : true
frequency : 0
scripts : http://10.100.100.76/APE_JSF/Source/mootools-core.js,http://10.100.100.76/APE_JSF/Source/Core/APE.js,http://10.100.100.76/APE_JSF/Source/Core/Events.js,http://10.100.100.76/APE_JSF/Source/Core/Core.js,http://10.100.100.76/APE_JSF/Source/Pipe/Pipe.js,http://10.100.100.76/APE_JSF/Source/Pipe/PipeProxy.js,http://10.100.100.76/APE_JSF/Source/Pipe/PipeMulti.js,http://10.100.100.76/APE_JSF/Source/Pipe/PipeSingle.js,http://10.100.100.76/APE_JSF/Source/Request/Request.js,http://10.100.100.76/APE_JSF/Source/Request/Request.Stack.js,http://10.100.100.76/APE_JSF/Source/Request/Request.CycledStack.js,http://10.100.100.76/APE_JSF/Source/Transport/Transport.longPolling.js,http://10.100.100.76/APE_JSF/Source/Transport/Transport.SSE.js,http://10.100.100.76/APE_JSF/Source/Transport/Transport.XHRStreaming.js,http://10.100.100.76/APE_JSF/Source/Transport/Transport.JSONP.js,http://10.100.100.76/APE_JSF/Source/Transport/Transport.WebSocket.js,http://10.100.100.76/APE_JSF/Source/Core/Utility.js,http://10.100.100.76/APE_JSF/Source/Core/JSON.js
baseUrl : http://10.100.100.76/APE_JSF
domain : 10.100.100.76
server : 10.100.100.76:5959

Running test : Setting document.domain
Running test : Checking APE.Config.baseUrl
APE JSF Version 1.0
Running test : Contacting APE Server
Connection etablished but something went wrong. Server response is

[{"time":"1423054314","raw":"ERR","data":{"code":"003","value":"BAD_CMD"}}]

Something went wrong. If you can't fix it by yourself post a message on the newsgroups with the output below or join our IRC channel

**Comment by lcharette** [comment]
The version number is fine. I remember the version number was not changed. You can see this in /Source/Core/Ape.js

As for the error, check server log and make sure you updated the server side scripts.

**Comment by symsun** [comment]
Thanks. Now both the server and client are working.

Is there a way to get rid of the frequency getting added to the ape server url. Can this be made optional.

http://0.ape.local.ape-project.org:6969/
http://1.ape.local.ape-project.org:6969/
http://2.ape.local.ape-project.org:6969/

**Comment by lcharette** [comment]
No, this is required for sessions support. You could edit the JSF source code if you don't plan on using sessions support. 

Envoyé de mon iPhone
